api/comment_threads.rb

```diff
@@ -27,9 +27,12 @@
     error 404, [t(:requested_object_not_found)].to_json
   end
 
-  if params["user_id"] and bool_mark_as_read
+  if params["user_id"]
     user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
-    user.mark_as_read(thread) if user
+  end
+
+  if user and bool_mark_as_read
+    user.mark_as_read(thread)
   end
```

Comment:
@wajeeha-khalid Yes, we keep track of read status for each user for each thread. 

However, the concern that @jcdyer raises is whether userX can update the read status of userY by simply passing userY's `user_id` instead of his own.  My response to that is that the Comments Service is currently an internal service and not accessible through a public interface.  Our defense and assumption is that the Mobile and Web Discussion APIs validate all parameters that get passed to the Comments Service.